chain/reorg_test.go

```diff
@@ -1,71 +1,141 @@
 package chain_test
 
 import (
+	"context"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	"github.com/filecoin-project/go-filecoin/chain"
 	th "github.com/filecoin-project/go-filecoin/testhelpers"
 	tf "github.com/filecoin-project/go-filecoin/testhelpers/testflags"
 	"github.com/filecoin-project/go-filecoin/types"
 )
 
-func TestIsReorg(t *testing.T) {
+func TestIsReorgFork(t *testing.T) {
 	tf.UnitTest(t)
+	dstP := initDSTParams()
+	ctx, blockSource, chainStore := setupGetAncestorTests(t, dstP)
+	old, new, common := getForkOldNewCommon(ctx, t, chainStore, blockSource, dstP)
+	assert.True(t, chain.IsReorg(old, new, common))
+}
+func TestIsReorgPrefix(t *testing.T) {
+	tf.UnitTest(t)
+	dstP := initDSTParams()
+	ctx, blockSource, chainStore := setupGetAncestorTests(t, dstP)
+	old, new, common := getPrefixOldNewCommon(ctx, t, chainStore, blockSource, dstP)
+	assert.False(t, chain.IsReorg(old, new, common))
+}
+
+func TestIsReorgSubset(t *testing.T) {
+	tf.UnitTest(t)
+	dstP := initDSTParams()
+	ctx, blockSource, chainStore := setupGetAncestorTests(t, dstP)
+	old, new, common := getSubsetOldNewCommon(ctx, t, chainStore, blockSource, dstP)
+	assert.False(t, chain.IsReorg(old, new, common))
+}
+
+func TestReorgDiffFork(t *testing.T) {
+	tf.UnitTest(t)
+	dstP := initDSTParams()
+	ctx, blockSource, chainStore := setupGetAncestorTests(t, dstP)
+	old, new, common := getForkOldNewCommon(ctx, t, chainStore, blockSource, dstP)
+
+	dropped, added, err := chain.ReorgDiff(old, new, common)
+	assert.NoError(t, err)
+	assert.Equal(t, uint64(11), dropped)
+	assert.Equal(t, uint64(14), added)
+}
+
+func TestReorgDiffSubset(t *testing.T) {
+	tf.UnitTest(t)
+	dstP := initDSTParams()
+	ctx, blockSource, chainStore := setupGetAncestorTests(t, dstP)
+	old, new, common := getSubsetOldNewCommon(ctx, t, chainStore, blockSource, dstP)
+
+	dropped, added, err := chain.ReorgDiff(old, new, common)
+	assert.NoError(t, err)
+	assert.Equal(t, uint64(1), dropped)
+	assert.Equal(t, uint64(1), added)
+}
+
+// getForkOldNewCommon
```

Comment:
Can we elaborate on this method more in the commnet? 

It looks like it grows the OG chain by 10 blocks, forks and grows the fork by 10, then grows the OG chain by 14 blocks. These numbers (10 & 14) seem like they might make good parameters to the method, what do you think? Also, is there anything significant about the numbers 10 and 14?